qemu/tests/multi_disk.py

```diff
@@ -346,31 +370,39 @@ def _get_disk_index(session, image_size, disk_indexs):
                 key_word = params["check_result_key_word"]
                 output = session.cmd_output(cmd)
                 if key_word not in output:
-                    raise error.TestFail("Files on guest os root fs and disk "
-                                         "differ")
+                    raise exceptions.TestFail("Files on guest os root "
+                                              "fs and disk differ")
 
+            error_context.context("Umount those disks", logging.info)
             if params.get("umount_command"):
-                cmd = params.get("show_mount_cmd")
-                output = session.cmd_output(cmd)
-                disks = re.findall(re_str, output)
-                disks.sort()
-                for disk in disks:
-                    disk = disk.strip()
-                    error.context("Unmounting disk: %s..." % disk)
-                    cmd = params.get("umount_command") % (disk, disk)
-                    session.cmd(cmd)
-    finally:
-        cmd = params.get("show_mount_cmd")
-        if cmd:
-            try:
-                output = session.cmd_output(cmd)
-                disks = re.findall(re_str, output)
-                disks.sort()
-                for disk in disks:
-                    error.context("Unmounting disk: %s..." % disk)
-                    cmd = params["umount_command"] % (disk, disk)
-                    session.cmd(cmd)
-            except Exception, err:
-                logging.warn("Get error when cleanup, '%s'", err)
+                try:
+                    for i in xrange(stg_image_num):
+                        if params.get("os_type") == "linux":
+                            partname = disk_indexs[i] + "1"
+                            devname = os.path.join("/dev", disk_indexs[i])
```

Comment:
@vivianQizhu Thanks. Will have a try. And since this patch depends on https://github.com/avocado-framework/avocado-vt/pull/887, will update it after the dependent one is merged.